File: system-test/java/src/main/java/com/mycompany/myshop/testkit/driver/adapter/api/client/controllers/HealthController.java
package com.mycompany.myshop.testkit.driver.adapter.api.client.controllers;

import com.mycompany.myshop.testkit.driver.adapter.api.client.dtos.errors.ProblemDetailResponse;
import com.mycompany.myshop.testkit.driver.adapter.shared.client.http.JsonHttpClient;
import com.mycompany.myshop.testkit.common.Result;

public class HealthController {
    private static final String ENDPOINT = "/health";

    private final JsonHttpClient<ProblemDetailResponse> httpClient;

    public HealthController(JsonHttpClient<ProblemDetailResponse> httpClient) {
        this.httpClient = httpClient;
    }

    public Result<Void, ProblemDetailResponse> checkHealth() {
        return httpClient.get(ENDPOINT);
    }
}

File: system-test/dotnet/Driver.Adapter/Api/Client/Controllers/HealthController.cs
using Common;

using Driver.Adapter.Shared.Client.Http;

using Driver.Adapter.Api.Client.Dtos.Errors;



namespace Driver.Adapter.Api.Client.Controllers;



public class HealthController

{

    private const string Endpoint = "/health";



    private readonly JsonHttpClient<ProblemDetailResponse> _httpClient;



    public HealthController(JsonHttpClient<ProblemDetailResponse> httpClient)

    {

        _httpClient = httpClient;

    }



    public Task<Result<VoidValue, ProblemDetailResponse>> CheckHealthAsync()

        => _httpClient.GetAsync(Endpoint);

}





